logging/Logger_unittest.cpp

```diff
@@ -0,0 +1,164 @@
+
+        file.close();
+        filename = "logs/" + filename;
+        std::remove(filename.c_str());
+        rmdir("logs");
+    }
+
+    TEST(Logging, vector_specification_level_in_file)
+    {
+        Logger logger;
+        std::string filename = TimeFormatter::get_time("YYYY-MM-DD-1.log");
+        std::vector<LogLevel> levels = {LogLevel::DEBUG, LogLevel::INFO, LogLevel::WARNING, LogLevel::ERROR, LogLevel::FATAL};
+
+        EXPECT_EQ(logger.get_display_specification_level_in_file(), levels);
+        logger.set_display_specification_level_in_file(LogLevel::DEBUG);
+        levels.push_back(LogLevel::DEBUG);
+        EXPECT_EQ(logger.get_display_specification_level_in_file(), levels);
+        logger.unset_display_specification_level_in_file(LogLevel::FATAL);
+        levels.erase(levels.end() - 2);
+        EXPECT_EQ(logger.get_display_specification_level_in_file(), levels);
+
+        filename = "logs/" + filename;
+        std::remove(filename.c_str());
+        rmdir("logs");
+    }
+
+    TEST(Logging, vector_specification_level_in_console)
+    {
+        Logger logger;
+        std::string filename = TimeFormatter::get_time("YYYY-MM-DD-1.log");
+        std::vector<LogLevel> levels = {LogLevel::DEBUG, LogLevel::INFO, LogLevel::WARNING, LogLevel::ERROR, LogLevel::FATAL};
+
+        EXPECT_EQ(logger.get_display_specification_level_in_console(), levels);
+        logger.set_display_specification_level_in_console(LogLevel::DEBUG);
+        levels.push_back(LogLevel::DEBUG);
+        EXPECT_EQ(logger.get_display_specification_level_in_console(), levels);
+        logger.unset_display_specification_level_in_console(LogLevel::FATAL);
+        levels.erase(levels.end() - 2);
+        EXPECT_EQ(logger.get_display_specification_level_in_console(), levels);
+
+        filename = "logs/" + filename;
+        std::remove(filename.c_str());
+        rmdir("logs");
+    }
+
+    TEST(Logging, display_specification_level_in_file)
+    {
+        Logger logger;
+        FileAndFolderHandler ff_handler;
+        std::string filename = TimeFormatter::get_time("YYYY-MM-DD-1.log");
+
+        testing::internal::CaptureStdout();
+        logger.log(LogLevel::DEBUG, "test");
+        std::string output = testing::internal::GetCapturedStdout();
+
+        std::fstream file("logs/" + filename, std::ios::in);
+        std::string line;
+        std::getline(file, line);
+
+        EXPECT_NE(line.find("test"), std::string::npos);
+        EXPECT_NE(output.find("test"), std::string::npos);
+
+        logger.unset_display_specification_level_in_file(LogLevel::DEBUG);
+
+        testing::internal::CaptureStdout();
+        logger.log(LogLevel::DEBUG, "not_a_test");
+        output = testing::internal::GetCapturedStdout();
+        std::getline(file, line);
+
+        EXPECT_EQ(line.find("not_a_test"), std::string::npos);
+        EXPECT_NE(output.find("not_a_test"), std::string::npos);
+
+        // file.seekg(0, std::ios::beg);
```

Comment:
Why isn't this working ?

Reply:
No idea but it would be better to use function mocking to test the whole

Reply:
Are you talking about that ? https://google.github.io/googletest/gmock_for_dummies.html
Because if yes, I didn't understand how can I do that